src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -332,32 +295,28 @@ - (void)didFinishWithError:(NSError *)errorOrNil {
 // after this.
 // The first one (metadataHandler), when the response headers are received.
 // The second one (completionHandler), whenever the RPC finishes for any reason.
-- (void)invokeCallWithMetadataHandler:(GRPCEventHandler)metadataHandler
-                    completionHandler:(GRPCEventHandler)completionHandler {
-  AssertNoErrorInCall(grpc_call_invoke_old(_gRPCCall,
-                                           _completionQueue.unmanagedQueue,
-                                           (__bridge_retained void *)metadataHandler,
-                                           (__bridge_retained void *)completionHandler,
-                                           0));
+- (void)invokeCallWithMetadataHandler:(GRPCCompletionHandler)metadataHandler
+                    completionHandler:(GRPCCompletionHandler)completionHandler {
+  [_wrappedCall startBatch:@{@(GRPC_OP_RECV_INITIAL_METADATA): @YES} handleCompletion:metadataHandler];
+  [_wrappedCall startBatch:@{@(GRPC_OP_RECV_STATUS_ON_CLIENT): @YES} handleCompletion:completionHandler];
```

Comment:
The plan is that after this pull request is merged, we will split Call into 4 classes, for each type of streaming. Then we will be able to have batches with multiple operations. But the current Call class represents bidi streaming, which requires more separation of operations in different batches.

In this particular case, if we put both of those in the same batch, we won't have the initial metadata from the server until the call finishes and we also have the status.